packages/pigeon/lib/dart_generator.dart

```diff
@@ -107,8 +108,28 @@ String _makeGenericCastCall(TypeDeclaration type, String nullTag) {
       : '';
 }
 
-String _getArgumentName(NamedType field) =>
-    field.name.isEmpty ? 'arg' : field.name;
+String _getArgumentName(int count, NamedType field) =>
+    field.name.isEmpty ? 'arg$count' : field.name;
+
+String _spaceJoin(String x, String y) => '$x $y';
+
+Iterable<String> _getMethodArgumentNames(Method func) =>
+    indexMap(func.arguments, _getArgumentName);
+
+String _getMethodArgumentsSignature(
+  Method func,
+  Iterable<String> argNames,
+  String nullTag,
+) {
+  return func.arguments.isEmpty
+      ? ''
+      : map2(
```

Comment:
It looks like_spaceJoin is the only thing you ever use `map2` for. You could avoid introducing a non-standard collection operation for readers to go look up by just doing a `zip`+`map` instead. You can then also trivially inline _spaceJoin instead of defining it. So it becomes:

```
  zip(
    func.arguments
        .map((NamedType e) => _addGenericTypes(e.type, nullTag)),
    argNames,
  )
  .map((Tuple t) => 't.first t.second')
  .join(', ');
```

That's not only less code overall, but completely understandable locally (assuming someone knows `zip`, but that's a pretty standard method in collection utilities) instead of having to look at `map2` and `_spaceJoin`.

Reply:
Allowing `map` to take in multiple arguments is pretty standard, both scheme and clojure allow it (I think some others too).  In common-lisp it's called `mapcar` which we shouldn't use haha. 

While writing this code I realized there was some overlap in use between zip and map2 (since zip is a special case of map2).  The reason I kept map2 is because of performance, avoiding allocations/deallocations, and it's slightly easier to work with since you don't have to consider what `.first` and `.second` actually mean (That wouldn't be as big of a deal if Dart could deconstruct tuples like F#).

Reply:
> Allowing map to take in multiple arguments is pretty standard, both scheme and clojure allow it

FYI, I'm pretty sure you are dramatically overestimating the percentage of developers who have any meaningful exposure to functional programming languages.

I'm not saying we can't adopt functional approaches where it's useful but "it's a standard thing someone reading this code should know" *definitely* does not follow from "Scheme and Lisp allow it".

Reply:
Haha fair enough, my examples will always be biased by my experience.  The question of if it's a standard function is a bit of a red herring to me.  The important questions are the ones we ask of any function: Is it useful?  Is it understandable?  Is it clearly documented?

Reply:
FWIW python does this as well: https://docs.python.org/3/library/functions.html#map

Reply:
> it's slightly easier to work with since you don't have to consider what `.first` and `.second` actually mean

I'm not sure I'm following this argument. Are you saying that going and looking up `_spaceJoin` and figuring out what `x` and `y` actually mean is easier than looking at the tuple created in the previous statement to see which argument is `first` and which is `second`?

`x` and `y` are both less meaningful, and further removed in the code from the context, so I'm not sure how that would be easier.

Reply:
Yep, `x` and `y` are only needed to understand the abstraction of `_spaceJoin`.  After I understand what `_spaceJoin` does they are not required for understanding `_getMethodArgumentsSignature`.  Whereas with a zip call I have to reference a different point in code to know what `.first` and `.second` refer to.  I'd say map2 is always preferable to zip in Dart.